src/GAP.jl

```diff
@@ -419,6 +420,37 @@ function prompt()
     reset_GAP_ERROR_OUTPUT()
 end
 
+"""
+    GAP.randseed!([seed::Integer])
+
+Reseed GAP's global RNG with `seed`.
+
+The given `seed` must be a non-negative integer.
+When `seed` is not specified, a random seed is generated from Julia's global RNG.
+
+For a fixed seed, the stream of generated numbers is allowed to change between
+different versions of GAP.
+"""
+function randseed!(seed::Union{Integer,Nothing}=nothing)
+    if seed === nothing
+        seed = rand(UInt128)
+    end
+    if 0 <= seed < typemax(Int) ÷ 2
+        # avoid too small seeds by hashing them
+        seed = hash(seed, 0x886b5530f00038ef % UInt)
```

Comment:
What is the issue with "too small seeds" ? Doesn't this effectively introduce bias to the seeds, in that many "small" seeds are impossible (they are mapped to "big" ones by the hashing), but not all (as the hash can of course also output something "small")?

Reply:
Yes I will remove that. The reasonning (coming from the equivalent PR in Nemo) was to improve the quality of the random streams given two seeds which are not very "reciprocally random", after reading from [GMP's docs](https://gmplib.org/manual/Random-State-Seeding#Random-State-Seeding)

> The “quality” of the seed is the randomness of a given seed compared to the previous seed used

But IIUC, GAP's Mersenne Twister implementation is from the same authors as what `Random` uses (dSFMT) and might do a better job at handling small seeds. Also, in Nemo, I used a cryptographic hash, but simply using `hash` here doesn't necessarily gives a good distribution.